### PR TITLE
Scala 2.12 support

### PR DESCRIPTION
**Pull request by takezoe**

### CODE REVIEW

**Comment by takezoe** [comment]
- all libraries are updated to Scala 2.12 version
- dependency to rl and scala-inflector are dropped (put required code into Scalatra repository)
- only metrics-scala for Scala 2.12 is still snapshot, so I will update it later
- compilation and all tests are passed for Scala 2.11 and 2.12

Could someone review and release the snapshot version of Scalatra 2.5 for Scala 2.11 and 2.12?

File: project/build.scala [hunk 1]
```diff
@@ -333,21 +325,25 @@ object ScalatraBuild extends Build {
...
+    private val akkaVersion             = "2.4.12"
+    private val grizzledSlf4jVersion    = "1.3.0"
     private val atmosphereCompatVersion = "2.0.1"
     private val httpcomponentsVersion   = "4.5.2"
     private val jettyVersion            = "9.2.17.v20160517"
-    private val json4sVersion           = "3.3.0"
-    private val scalateVersion          = "1.7.1"
-    private val scalatestVersion        = "2.2.5"
-    private val specs2Version           = "3.7.3"
+    private val json4sVersion           = "3.5.0.RC1"
```
Comment (xuwei-k) [inline project/build.scala]: 3.5.0 is latest version

https://oss.sonatype.org/content/repositories/releases/org/json4s/json4s-core_2.12/

File: project/build.scala [hunk 2]
```diff
@@ -333,21 +325,25 @@ object ScalatraBuild extends Build {
...
+    private val scalateVersion          = "1.8.0-RC1"
```
Comment (seratch) [inline project/build.scala]: scalate 1.8.0 is available.

**Comment by seratch** [comment]
>only metrics-scala for Scala 2.12 is still snapshot, so I will update it later

metrics-scala is available on the maven central now.
- https://github.com/erikvanoosten/metrics-scala/pull/85#issuecomment-258612742
- http://search.maven.org/#search%7Cgav%7C1%7Cg%3A%22nl.grons%22%20AND%20a%3A%22metrics-scala_2.12%22

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 language: scala
 scala:
-  - 2.10.6
```
Comment (seratch) [inline .travis.yml]: Dropping 2.10 support is ok for me but we should mention it in the announcement of 2.5.0 release.

Comment (takezoe) [inline .travis.yml]: Yes, I think so. Dropping Java7 is too.

**Comment by takezoe** [comment]
@seratch @xuwei-k Thanks. All updated.

**Comment by xuwei-k** [comment]
🎉

**Comment by seratch** [comment]
Thanks for your great work @takezoe :bow: Let me merge this and I will publish 2.5.0-M1 within hours.

**Pull request merged**